### ISSUE TITLE
Is Vue-i18n fallback with decision map supported?

### ISSUE THREAD

**Issue by DktRemiDoolaeghe** [nuxt-modules/i18n#991]
I'm trying to use `Vue-i18n` ability to have a decision map when using a fallback locale, as described in [vue-i18n documentation](https://kazupon.github.io/vue-i18n/guide/fallback.html#explicit-fallback-with-decision-maps). This is the configuration I have used:

```js
['nuxt-i18n', {
  locales: [
    {
      code: 'fr-FR',
      file: 'fr-FR.js'
    },
    {
      code: 'en-GB',
      file: 'en-GB.js'
    },
    {
      code: 'en',
      file: 'en.js'
    },
    {
      code: 'fr-BE',
      file: 'fr-BE.js'
    },
    {
      code: 'nl-BE',
      file: 'nl-BE.js'
    }
  ],
  lazy: true,
  // https://i18n.nuxtjs.org/options-reference#strategy
  strategy: 'prefix',
  langDir: './static/i18n/lang/',
  detectBrowserLanguage: false,
  vueI18n: {
    fallbackLocale: {
      'en-GB': ['fr-FR']
    },
    silentFallbackWarn: true
  }
}]
```

At runtime, let's describe what happens. Let's consider that:
* `en-GB` locale is set in my app
* `my_key` is missing a translation for `en-GB` locale
* `my_key` has a translation for `fr-FR` locale

I was expecting that the `fr-FR` translation would be selected, to provide a fallback translation as it was missing in the `en-GB` locale, but it seems it is not the case.

After having a look on the source code, I think I found that this feature is not implemented, as demonstrated by this code extract from `src/templates/plugin.main.js`:

``` js
/** @type {import('@nuxt/types').Plugin} */
export default async (context) => {
  ...
  const loadAndSetLocale = async (newLocale, { initialSetup = false } = {}) => {
    ...
    // Load fallback locale.
    if (app.i18n.fallbackLocale && newLocale !== app.i18n.fallbackLocale) { // <-- HERE
      await loadLanguageAsync(context, app.i18n.fallbackLocale)
    }
    await loadLanguageAsync(context, newLocale)
  }
}
```

On runtime, at the line identified by the comment **HERE**, `app.i18n.fallbackLocale` evaluates to `{en-GB:["fr-FR"]}`. So the condition in the `if` can never be true with a decision map, and fallback locales cannot be loaded.

Is there any plan to provide this feature in `nuxt-i18n`? If no, can I (if I have time) provide a PR? If yes, have you any advice before I plan on working on the code?

Thanks

**Comment by rchl** [comment]
Looks like a missing feature (since I suppose it's a relatively new feature in `vue-i18n` also).

As far as working on a PR, you are very welcome. Check https://github.com/nuxt-community/i18n-module/blob/master/CONTRIBUTING.md

**Comment by DktRemiDoolaeghe** [comment]
I wish to push a feature branch and create a PR, but it seems I have no right to. Do I have something to do, to be able to push my feature branch?

**Comment by divine** [comment]
> I wish to push a feature branch and create a PR, but it seems I have no right to. Do I have something to do, to be able to push my feature branch?

You don't need a permission to do that.

Please have a read https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork

Thanks!

**Comment by DktRemiDoolaeghe** [comment]
I may miss something, but it seems I can't push for a reason I cannot understand.

![Capture d’écran 2020-12-09 094459](https://user-images.githubusercontent.com/51918193/101606360-9986e700-3a03-11eb-8015-9f2fc3509f7e.jpg)

I precise I can do a `git pull` with success.

Any idea?

**Comment by rchl** [comment]
You need to push to your forked remote, not to `origin` which you have set up to point to this original repo.

**Comment by DktRemiDoolaeghe** [comment]
Ok, thanks for your help. The PR has been created.

**Comment by rchl** [comment]
Was implemented through #992 and released in https://github.com/nuxt-community/i18n-module/releases/tag/v6.17.0.
